### PR TITLE
Client - Implemented foundry output

### LINKED ISSUE #262: Client - update foundry output
## Description

Implement deserialization from JSON object for foundry output type to be compatible with newest protocol changes.

### PR DESCRIPTION
**Pull request by matjazv**
# Description of change

- Fixes #262 
- Fixed memory management bug in JSON parser for alias and extended outputs
- Changed print functions for transaction, inputs and outputs

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

New unit tests implemented in `/client/api_json_parser/test_output_foundry.c`

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/client/api_json_parser/test_output_foundry.c [hunk 1]
```diff
@@ -0,0 +1,172 @@
+// Copyright 2020 IOTA Stiftungnative_tokens
```
Comment (oopsmonk) [inline tests/client/api_json_parser/test_output_foundry.c]: happy new year :P 
```suggestion
// Copyright 2022 IOTA Stiftungnative_tokens
```

File: src/client/api/json_parser/output_foundry.c [hunk 2]
```diff
@@ -2,12 +2,169 @@
...
+  // circulation supply
+  char circ_supply_str[STRING_NUMBER_MAX_CHARACTERS];
+  if (json_get_string(output_obj, JSON_KEY_CIRC_SUPPLY, circ_supply_str, STRING_NUMBER_MAX_CHARACTERS) != JSON_OK) {
+    printf("[%s:%d]: getting %s json string failed\n", __func__, __LINE__, JSON_KEY_CIRC_SUPPLY);
+    goto end;
+  }
+  circ_supply = uint256_from_str(circ_supply_str);
+
+  // maximum supply
+  char max_supply_str[STRING_NUMBER_MAX_CHARACTERS];
```
Comment (oopsmonk) [inline src/client/api/json_parser/output_foundry.c]: for reducing the stack size, both circ and max supply strings should use the same buffer.

**Review by oopsmonk — CHANGES_REQUESTED** [review]

File: tests/client/api_json_parser/test_output_foundry.c (hunk 1 above)
Comment (matjazv) [reply tests/client/api_json_parser/test_output_foundry.c]: Fixed.

File: src/client/api/json_parser/output_foundry.c (hunk 2 above)
Comment (matjazv) [reply src/client/api/json_parser/output_foundry.c]: Fixed.

File: src/core/utils/uint256.c [hunk 3]
```diff
@@ -178,7 +178,14 @@ int uint256_equal(uint256_t const *a, uint256_t const *b) {
...
 
-  return memcmp(a, b, sizeof(uint256_t));
+  // little endian comparison
+  for (int8_t i = 3; i >= 0; i--) {
+    if (a->bits[i] != b->bits[i]) {
+      return memcmp(&a->bits[i], &b->bits[i], sizeof(uint64_t));
+    }
+  }
+
+  return 0;
```
Comment (oopsmonk) [inline src/core/utils/uint256.c:189]: line#178 returning 0 indicates `a` is equal to `b`, it is not true.

**Review by oopsmonk — CHANGES_REQUESTED** [review]

File: src/core/utils/uint256.c (hunk 3 above)
Comment (matjazv) [reply src/core/utils/uint256.c:189]: Done.

**Review by oopsmonk — APPROVED** [review]

**Pull request merged**